pkg/cmd/roachtest/registry/operation_spec.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package registry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+)
+
+type OperationDependency int
+
+const (
+	OperationRequiresNodes OperationDependency = iota
+	OperationRequiresSqlConnection
+	OperationRequiresDatabaseSchema
+	OperationRequiresPopulatedDatabase
+)
+
+// OperationSpec is a spec for a roachtest operation.
+type OperationSpec struct {
+	Skip string // if non-empty, operation will be skipped
+	// When Skip is set, this can contain more text to be printed in the logs
+	// after the "--- SKIP" line.
+	SkipDetails string
+
+	Name string
+	// Owner is the name of the team responsible for signing off on failures of
+	// this test that happen in the release process. This must be one of a limited
+	// set of values (the keys in the roachtestTeams map).
+	Owner Owner
+	// The maximum duration the test is allowed to run before it is considered
+	// failed. If not specified, the default timeout is 10m before the test's
+	// associated cluster expires. The timeout is always truncated to 10m before
+	// the test's cluster expires.
+	Timeout time.Duration
+
+	// CompatibleClouds is the set of clouds this test can run on (e.g. AllClouds,
+	// OnlyGCE, etc). Must be set.
+	CompatibleClouds CloudSet
+
+	// Suites is the set of suites this test is part of (e.g. Nightly, Weekly,
+	// etc). Must be set, even if empty (see ManualOnly).
+	Suites SuiteSet
+
+	// Dependency specify the types of resources required for this roachtest
+	// operation to work. This will be used to match this operation up with
+	// eligible clusters to run.
+	//
+	// TODO(bilal): Unused.
+	Dependency OperationDependency
+
+	// CanRunConcurrently specifies whether multiple instances of this operation
+	// can run in parallel.
+	//
+	// TODO(bilal): Unused.
+	CanRunConcurrently bool
+
+	// RequiresLicense indicates that the test requires an
+	// enterprise license to run correctly. Use this to ensure
+	// tests will fail-early if COCKROACH_DEV_LICENSE is not set
+	// in the environment.
```

Comment:
Documentation copy pasted from `TestSpec`, and I don't think it makes much sense in the context of operations. In tests, this will ensure that the cluster has an enterprise license when it is provisioned. Since we are using an existing cluster for operations, I'm not sure what the semantics of this field are. We should either drop it if it's not being used, or update the documentation.

Reply:
Updated comment. I thought it was still worthwhile as some operations could do an IMPORT or a RESTORE, and we can self-select operations out if our target cluster isn't enterprise-y. Also allows for better nesting within roachtests.

Reply:
Removed.